modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java

```diff
@@ -760,6 +762,35 @@ public void testGetObjectEntryHierarchy() throws Exception {
 			() -> _objectEntryService.getObjectEntry(rootObjectEntryId));
 	}
 
+	@Test
+	public void testGetOrAddEmptyObjectEntry() throws Exception {
+		try (SafeCloseable safeCloseable =
+				LazyReferencingThreadLocal.setEnabledWithSafeCloseable(true)) {
+
+			_setUser(_guestUser);
+
+			AssertUtils.assertFailure(
+				PrincipalException.MustHavePermission.class,
+				StringBundler.concat(
+					"User ", _guestUser.getUserId(),
+					" must have ADD_OBJECT_ENTRY permission for ",
+					_objectDefinition.getResourceName(), " ",
+					_guestUser.getGroupId()),
+				() -> _objectEntryService.getOrAddEmptyObjectEntry(
+					_objectDefinition.getExternalReferenceCode(),
+					_guestUser.getGroupId(), _guestUser.getUserId(),
+					_objectDefinition.getObjectDefinitionId()));
+
+			_setUser(_adminUser);
+
+			Assert.assertNotNull(
+				_objectEntryService.getOrAddEmptyObjectEntry(
+					_objectDefinition.getExternalReferenceCode(),
+					_adminUser.getGroupId(), _adminUser.getUserId(),
+					_objectDefinition.getObjectDefinitionId()));
```

Comment:
The externalReferenceCode is from the object entry, so it should be a random string.